pkg/messagebroker/kafka.go

```diff
@@ -392,7 +392,7 @@ func (k *KafkaBroker) SendMessage(ctx context.Context, request SendMessageToTopi
 		TopicPartition: kafkapkg.TopicPartition{Topic: &topicN, Partition: kafkapkg.PartitionAny},
 		Value:          request.Message,
 		Key:            []byte(request.OrderingKey),
-		Headers:        kHeaders,
+		Headers:        carrier,
```

Comment:
isn't carrier updating the same set of headers ?

Reply:
Yes but we pass the kHeaders object to carrier and not it's reference. Hence the original object does not get updated.